monkey/monkey_island/cc/repositories/utils/agent_configuration_schema_compiler.py

```diff
@@ -1,48 +1,130 @@
             raise RuntimeError(err)
 
+    @staticmethod
+    def _pydantic_schema_generation_override():
+        # See https://github.com/pydantic/pydantic/issues/5106
+        from pydantic.schema import default_ref_template
+
+        def model_process_schema_internal(
+            model,
+            *,
+            by_alias: bool = True,
+            generate_descriptions: bool = True,  # noqa: F841
+            model_name_map,
+            ref_prefix=None,
+            ref_template: str = default_ref_template,  # type: ignore
+            known_models=None,
+            field=None,
+        ):
+            from inspect import signature
+
+            known_models = known_models or set()
+            from pydantic.utils import lenient_issubclass
+
+            if lenient_issubclass(model, Enum):
+                model = cast(Type[Enum], model)
+                from pydantic.schema import enum_process_schema
+
+                s = enum_process_schema(model, field=field)
+                return s, {}, set()
+            model = cast(Type["BaseModel"], model)
+            s = {"title": model.__config__.title or model.__name__}
+            known_models.add(model)
+            from pydantic.schema import model_type_schema
+
+            m_schema, m_definitions, nested_models = model_type_schema(
+                model,
+                by_alias=by_alias,
+                model_name_map=model_name_map,
+                ref_prefix=ref_prefix,
+                ref_template=ref_template,
+                known_models=known_models,
+            )
+            s.update(m_schema)
+            schema_extra = model.__config__.schema_extra
+            if callable(schema_extra):
+                if len(signature(schema_extra).parameters) == 1:
+                    schema_extra(s)
+                else:
+                    schema_extra(s, model)
+            else:
+                s.update(schema_extra)
+            return s, m_definitions, nested_models
+
+        pydantic.schema.model_process_schema = model_process_schema_internal
+
     def _add_plugins(self, schema: Dict[str, Any]) -> Dict[str, Any]:
         schema = self._add_properties_field_to_plugin_types(schema)
-        schema = self._add_non_plugin_exploiters(schema)
+
+        # Hard coded plugins need to be added manually until they are refactored into
+        # proper plugins
+        schema = self._add_hard_coded_plugins(schema)
+
         config_schemas = deepcopy(
             self._agent_plugin_repository.get_all_plugin_configuration_schemas()
         )
 
         for plugin_type in config_schemas.keys():
             for plugin_name in config_schemas[plugin_type].keys():
                 config_schema = config_schemas[plugin_type][plugin_name]
+                plugin_manifest = self._agent_plugin_repository.get_all_plugin_manifests()[
+                    plugin_type
+                ][plugin_name]
+                config_schema.update(plugin_manifest.dict(simplify=True))
```

Comment:
Why are we adding the manifest to the configuration schema? Wouldn't that put information like `safe` into the config schema? `safe` is not configurable.

Reply:
Discussed on webex.